src/pages/CreateAssignment/CreateAssignment.js

```diff
@@ -12,7 +12,9 @@ import DatePicker from "components/DatePicker/DatePicker.js";
 import WordGroupSelector from "../../components/WordGroupSelector/WordGroupSelector";
 import SectionSelector from "../../components/SectionSelector/SectionSelector";
 
-function CreateAssignment({ firebase }) {
+function CreateAssignment(props) {
+  const { firebase } = props;
+  const { existingAssignment } = props?.location.state;
```

Comment:
is there no other way to go about having optional props in hooks?

Reply:
wdym?

Reply:
ig a q i have from this is does setting firebase = props work if there's existingAssignment and firebase as props? what does the props obj look like on rendering?

Reply:
Ohhh I see what you're trying to say -- this isn't storing "props" the object as a firebase object, it's destructuring the object so that the first field is being stored as firebase (which is always the "firebase" field)

existingAssignment on the other hand is being destructured from within props.location.state